### PR TITLE
Fixes #2884

### LINKED ISSUE #2884: Repeated column when use encrypt with JPA + Springboot
### Which version of ShardingSphere did you use?

4.0.0-RC2-SNAPSHOT

### Which project did you use? Sharding-JDBC or Sharding-Proxy?

Sharding-JDBC

### Expected behavior

Insert encrypt successfully.

### Actual behavior

```
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Column 'user_name_plain' specified twice
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
	at com.mysql.jdbc.Util.getInstance(Util.java:408)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:943)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2490)
	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1858)
	at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2079)
	at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2013)
	at com.mysql.jdbc.PreparedStatement.executeLargeUpdate(PreparedStatement.java:5104)
	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:1998)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
	at org.apache.shardingsphere.shardingjdbc.jdbc.core.statement.EncryptPreparedStatement.executeUpdate(EncryptPreparedStatement.java:120)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:208)
	... 23 more
```

### Reason analyze (If you can)

Logic SQL: `insert into t_user (assisted_query_pwd, pwd, user_name, user_name_plain, user_id) values (?, ?, ?, ?, ?)`

actual SQL: `insert into t_user (assisted_query_pwd, pwd, user_name, user_name_plain, user_id, assisted_query_pwd, user_name_plain) values (?, ?, ?, ?, ?, ?, ?) `

SQL generated by JPA has included `user_name_plain `, but encrypt add `user_name_plain` again.

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

clone example code.
run `sharding-jdbc-example/other-feature-example/encrypt-example/encrypt-spring-boot-example/src/main/java/org/apache/shardingsphere/example/encrypt/table/spring/boot/SpringBootExample.java`

### Example codes for reproduce this issue (such as a github link).

https://github.com/apache/incubator-shardingsphere-example

### PR DESCRIPTION
**Pull request by RaigorJiang**
Fixes #2884

Changes proposed in this pull request:
- If there are related fields in the original SQL, do not add them;

### CODE REVIEW

**Comment by RaigorJiang** [comment]
OK, I'll add some test cases.

**Comment by KomachiSion** [comment]
I will review this fix as soon as possible, Thanks @RaigorJiang .

File: sharding-core/sharding-core-rewrite/src/main/java/org/apache/shardingsphere/core/rewrite/feature/encrypt/token/generator/impl/AssistQueryAndPlainInsertColumnsTokenGenerator.java [hunk 1]
```diff
@@ -52,9 +54,13 @@ protected boolean isGenerateSQLTokenForEncrypt(final SQLStatementContext sqlStat
         Collection<InsertColumnsToken> result = new LinkedList<>();
         Optional<EncryptTable> encryptTable = getEncryptRule().findEncryptTable(sqlStatementContext.getTablesContext().getSingleTableName());
         Preconditions.checkState(encryptTable.isPresent());
+        Set<String> columnNames = new HashSet(((InsertStatement) sqlStatementContext.getSqlStatement()).getColumnNames());
         for (ColumnSegment each : ((InsertStatement) sqlStatementContext.getSqlStatement()).getColumns()) {
             List<String> columns = getColumns(encryptTable.get(), each);
             if (!columns.isEmpty()) {
+                if (columnNames.contains(columns.get(0))) {
+                    continue;
+                }
```
Comment (KomachiSion) [inline sharding-core/sharding-core-rewrite/src/main/java/org/apache/shardingsphere/core/rewrite/feature/encrypt/token/generator/impl/AssistQueryAndPlainInsertColumnsTokenGenerator.java]: I think move this logic into `getColumns` function will more reabable, is it?

File: sharding-jdbc/sharding-jdbc-core/src/test/resources/jdbc_init.sql [hunk 2]
```diff
@@ -22,4 +22,5 @@ CREATE TABLE IF NOT EXISTS t_order_item_1 (item_id INT NOT NULL, order_id INT NO
 CREATE TABLE IF NOT EXISTS t_config (id INT NOT NULL, status VARCHAR(45) NULL, PRIMARY KEY (id));
 CREATE TABLE IF NOT EXISTS t_global (id INT NOT NULL, status VARCHAR(45) NULL, PRIMARY KEY (id));
 CREATE TABLE IF NOT EXISTS t_encrypt (id INT NOT NULL AUTO_INCREMENT, cipher_pwd VARCHAR(45) NULL, plain_pwd VARCHAR(45), PRIMARY KEY (id));
-CREATE TABLE IF NOT EXISTS t_query_encrypt (id INT NOT NULL AUTO_INCREMENT, cipher_pwd VARCHAR(45) NULL, assist_pwd VARCHAR(45) NULL, PRIMARY KEY (id));
\ No newline at end of file
+CREATE TABLE IF NOT EXISTS t_query_encrypt (id INT NOT NULL AUTO_INCREMENT, cipher_pwd VARCHAR(45) NULL, assist_pwd VARCHAR(45) NULL, PRIMARY KEY (id));
+CREATE TABLE IF NOT EXISTS t_query_and_plain_encrypt (id INT NOT NULL AUTO_INCREMENT, cipher_pwd VARCHAR(45) NULL, plain_pwd VARCHAR(45) NULL, assist_pwd VARCHAR(45) NULL, PRIMARY KEY (id));
```
Comment (KomachiSion) [inline sharding-jdbc/sharding-jdbc-core/src/test/resources/jdbc_init.sql]: Please end file with a blank line

File: sharding-core/sharding-core-rewrite/src/main/java/org/apache/shardingsphere/core/rewrite/feature/encrypt/token/generator/impl/AssistQueryAndPlainInsertColumnsTokenGenerator.java (hunk 1 above)
Comment (RaigorJiang) [reply sharding-core/sharding-core-rewrite/src/main/java/org/apache/shardingsphere/core/rewrite/feature/encrypt/token/generator/impl/AssistQueryAndPlainInsertColumnsTokenGenerator.java]: Yes, moved.

**Pull request merged**